lucene/queries/src/test/org/apache/lucene/queries/intervals/TestIntervals.java

```diff
@@ -393,11 +400,16 @@ public void testNesting2() throws IOException {
     assertNull(getMatches(source, 0, "field1"));
     MatchesIterator it = getMatches(source, 1, "field1");
     assertMatch(it, 6, 21, 41, 118);
+    assertNull("Conjunction intervals don't yield query so far. dunno y",it.getQuery());
```

Comment:
We should probably fix this before committing?

Reply:
I don't feel like it's blocker. Since, now master just throws exception, and this PR at least highlight something, and actually quite a lot. It seems like highlighting grasps into terms intervals well. 
From the other side, if it returns something like plain BQ of TQ, it might cause more coarse highlighting, or you propose to return IntervalQuery here?  
Thanks for your suggestions above. I'm completely lost in this tunnel.

Reply:
Actually I think it's fine, because `IntervalWeight.matches()` always wraps these to return the correct query - you can just remove this line.